tests/test_check_functions.py

```diff
@@ -0,0 +1,66 @@
+import json
+import unittest
+
+from spinedb_api.db_cache import DBCache, ParameterValueItem
+from spinedb_api.exception import SpineIntegrityError
+
+from spinedb_api.check_functions import replace_parameter_values_with_list_references
+
+
+class TestCheckFunctions(unittest.TestCase):
+    def setUp(self):
+        self.data = [
+            (bool, (b'"TRUE"', b'"FALSE"', b'"T"', b'"True"', b'"False"'), (b'true', b'false')),
+            (int, (b'32',), (b'42', b'-2')),
+            (str, (b'"FOO"', b'"bar"'), (b'"foo"', b'"Bar"', b'"BAZ"')),
+        ]
+        self.par_defns = {
```

Comment:
Why abbreviate names like this? `parameter_definitions` is much clearer. Same for `par_val_lists` and `list_vals` below. Personally, I'd even prepend these with underscores `_parameter_definitions` but if you said I am being too pedantic here I would wholeheartedly agree :)

Reply:
That's on me.  I was getting annoyed of the long lines going beyond my terminal width, and since these are tests I took the liberty.  But you are right, I'll use the full names.